internal/cmdliveinit/cmdliveinit.go

```diff
@@ -182,54 +187,70 @@ func (c *ConfigureInventoryInfo) Run(ctx context.Context) error {
 	return nil
 }
 
-// Run fills in the inventory object values into the Kptfile.
-func updateKptfile(p *pkg.Pkg, inv *kptfilev1.Inventory, force bool) error {
-	const op errors.Op = "cmdliveinit.updateKptfile"
-	// Read the Kptfile io io.dir
+// createRGFile fills in the inventory object values into the resourcegroup object and writes to file storage.
+func createRGFile(p *pkg.Pkg, inv *kptfilev1.Inventory, filename string, force bool) error {
+	const op errors.Op = "cmdliveinit.updateResourceGroup"
+	// Read the resourcegroup object io io.dir
+	rg, err := p.ReadRGFile(filename)
+	if err != nil && !goerrors.Is(err, os.ErrNotExist) {
+		return errors.E(op, p.UniquePath, err)
+	}
+
+	// Read the Kptfile to ensure that inventory information is not in Kptfile either.
 	kf, err := p.Kptfile()
 	if err != nil {
 		return errors.E(op, p.UniquePath, err)
 	}
-	// Validate the inventory values don't already exist
+	// Validate the inventory values don't exist in Kptfile.
 	isEmpty := kptfileInventoryEmpty(kf.Inventory)
 	if !isEmpty && !force {
+		return errors.E(op, p.UniquePath, &InvInKfExistsError{})
+	}
+	// Set the Kptfile inventory to be nil since we force write to resourcegroup instead.
+	kf.Inventory = nil
```

Comment:
In `kpt live init`, if Kptfile.inventory is not empty, we shall make it backward compatible and not wipe out the Kptfile inventory. 
In such case, kpt shall warn users that the inventory already exist and they shall run `kpt live migrate`

Reply:
This is already handled by the current logic. The following error message is throw when a user runs `kpt live init` with a Kptfile that contains an inventory:
```
Error: Inventory information has already been added to the package Kptfile object. Please consider migrating to a standalone resourcegroup object using the 'kpt live migrate' command.
```

We should still empty out the Inventory if a user uses the `--force` flag, ie: `kpt live init --force` on a package where a Kptfile contains the inventory information already.

Also, `kpt live migrate` reuses code in `kpt live apply`, and internally, we provide the force option (but reuse existing Kptfile inventory) to handle migrating inventory from Kptfile to resourcegroup.

Reply:
It seems like we are updating the error message here to refer to changes that isn't included in this PR. It is unclear to me if merging this PR by itself will leave things in a bad state that would block releases.

Reply:
Instead of using a feature branch, this PR has been reworked to gate the new additions behind an arg. The existing flows for users should be unchanged, and the e2e tests have been re-enabled.